app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        render_index_response
+      end
+
+      def show
+        render json: note, status: :ok, serializer: NoteSerializer
+      end
+
+      private
+
+      def note
+        Note.find(params.require(:id))
+      end
+
+      def render_index_response
+        if invalid_order?
+          return render_invalid_parameter_error(I18n.t('errors.messages.invalid_order'))
+        end
+        if invalid_note_type?
+          return render_invalid_parameter_error(I18n.t('errors.messages.invalid_note_type'))
+        end
+
```

Comment:
Siempre deberíamos apuntar a  refactorizamos este tipo de métodos y favorecer utilizar la versión corta del if,  para que nos quede una estructura mucho más simple de este estilo:

```suggestion
    return render_caso_triste_1 if error_1?
    return render_caso_triste_2 if error_2?

    render_caso_feliz
```

Además nuevos casos son fáciles de agregar y no nos fuerzan a hacer un refactor más adelante solo para agregarlos